ci/pingcap_tidb_operator_build_kind.groovy

```diff
@@ -47,104 +134,56 @@ def call(BUILD_BRANCH, CREDENTIALS_ID, CODECOV_CREDENTIALS_ID) {
 						}
+					}
 
+					stage("Check") {
 						ansiColor('xterm') {
-						sh """
-						export GOPATH=${WORKSPACE}/go
-						export PATH=${WORKSPACE}/go/bin:\$PATH
-						if ! hash hg 2>/dev/null; then
-							sudo yum install -y mercurial
-						fi
-						hg --version
-						make check-setup
-						make check
-						if [ ${BUILD_BRANCH} == "master" ]
-						then
-							make test GO_COVER=y
-							curl -s https://codecov.io/bash | bash -s - -t ${CODECOV_TOKEN} || echo'Codecov did not collect coverage reports'
-						else
-							make test
-						fi
-						make
-						make e2e-build
-						"""
+							sh """
+							export GOPATH=${WORKSPACE}/go
+							export PATH=${WORKSPACE}/go/bin:\$PATH
+							make check-setup
+							make check
+							"""
 						}
 					}
-				}
-				stash excludes: "${PROJECT_DIR}/vendor/**,${PROJECT_DIR}/deploy/**", includes: "${PROJECT_DIR}/**", name: "tidb-operator"
-			}
-		}
 
-		node('k8s-kind') {
-			deleteDir()
-			unstash 'tidb-operator'
-
-			dir("${PROJECT_DIR}"){
-				stage('start run operator e2e test'){
-					ansiColor('xterm') {
-					sh """#/usr/bin/env bash
-					echo "info: setup go"
-					export PATH=\$PATH:/usr/local/go/bin
-					echo "==== Start Build Environment ===="
-					echo "BASH VERSION: \$BASH_VERSION"
-					echo "PATH: \$PATH"
-					echo "PROJECT_DIR: ${PROJECT_DIR}"
-					if ! command -v go &>/dev/null; then
-						echo "error: go is required"
-						exit 1
-					fi
-					go version
-					if ! command -v docker &>/dev/null; then
-						echo "error: docker is required"
-						exit 1
-					fi
-					docker version
-					echo "==== End Build Environment ===="
-					declare -A clusters
-					clusters['kind']='127.0.0.1:5000'
-					clusters['kind2']='127.0.0.2:5000'
-					clusters['kind3']='127.0.0.3:5000'
-					clusters['kind4']='127.0.0.4:5000'
-					while true
-					do
-						for cluster in \${!clusters[*]}; do
-							lockfile=/k8s-locks/\$cluster
-							if [ -e \$lockfile ]; then
-								prId=`cat \$lockfile`
-								if [ "${ghprbPullId}" == "\$prId" ]; then
-									clusterName=\$cluster
-									echo "####### branch ${ghprbPullId}/${BUILD_BRANCH} get env:\$cluster ########"
-									echo "####### if you want to debug,please exec the command: export KUBECONFIG=`/root/go/bin/kind get kubeconfig-path --name=\$cluster ` in jenkins slave node ########"
-										break 2
-								fi
-							fi
-						done
-
-						for cluster in \$(shuf -e \${!clusters[*]}); do
-							lockfile=/k8s-locks/\$cluster
-							if [ ! -e \$lockfile ]; then
-									touch \$lockfile
-									echo ${ghprbPullId} > \$lockfile
-									clusterName=\$cluster
-									echo "####### branch ${ghprbPullId}/${BUILD_BRANCH} get env:\$cluster #######"
-									echo "####### if you want to debug,please exec the command: export KUBECONFIG=`/root/go/bin/kind get kubeconfig-path --name=\$cluster ` in jenkins slave node ########"
-									break 2
+					stage("Build and Test") {
+						ansiColor('xterm') {
```

Comment:
What shell is used by default in Jenkins? Shall we make it explicit?

Reply:
`sh`. I think it is not necessary because we don't write complex inline scripts in Jenkins pipeline anymore.